api/soap/mc_api.php

```diff
@@ -43,6 +43,11 @@
  */
 class SoapObjectsFactory {
 	/**
+	 * @var bool true: SOAP API, false: REST API
+	 */
+	static public $soap = true;
```

Comment:
I've renamed this class and some of its methods so it is not Soap specific, but ApiObjectsFactory.  I kept methods that are providing added value or may behave differently based on API type.  More APIs may be added to this later to handle distinction between REST and SOAP.